### PR TITLE
Resolves issue #18

### PR DESCRIPTION
**Pull request by peter279k**
# Changed log

- Resolves issue #18.
- There are know issues:
   - The Infection should be upgraded because the version is complained during analysis running.
   - Some tests are failed in the `PHP 7.2` version when running the `make ci-test --keep-going` command.
   - Due to the above reason, the GitHub action is expected to be failed.
   - The coveralls steps that I will not migrate because it needs the GitHub token. Please refer the [reference](https://github.com/php-coveralls/php-coveralls#github-actions).
- The latest GitHub action log is available [here](https://github.com/peter279k/include-interceptor/actions/runs/3654532272).

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+on: [ push, pull_request ]
+
+jobs:
+  run:
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      matrix:
+        operating-system: [ ubuntu-latest ]
+        php-versions: [ '7.2', '7.3', '7.4' ]
```
Comment (maks-rafalko) [inline .github/workflows/ci.yaml:11]: I think we should test starting from PHP 8.0, this is what Infection itself is currently being tested on

related to https://github.com/infection/infection/pull/1765

**Comment by maks-rafalko** [comment]
Sorry for the late review. Regarding coveralls, would this work? (our another repository already uses it)

https://github.com/infection/extension-installer/blob/c580e7571ea339d6662acda32984134711f55b58/.github/workflows/ci.yaml#L57-L60

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        with:
+          php-version: ${{ matrix.php-versions }}
+          extensions: ast
+          coverage: pcov
+
+      - name: Validate composer.json and composer.lock
+        run: composer validate
+
+      - name: Install dependencies
+        run: COMPOSER_ROOT_VERSION=0.2.5 composer install --prefer-dist --no-progress
```
Comment (peter279k) [inline .github/workflows/ci.yaml:29]: Adding the `COMPOSER_ROOT_VERSION` environment variable to resolve the following error:

```
  Problem 1
    - Root composer.json requires infection/infection ^0.19.0 -> satisfiable by infection/infection[0.19.0, 0.19.1, 0.19.2].
    - infection/infection[0.19.0, ..., 0.19.2] require infection/include-interceptor ^0.2.4 -> satisfiable by infection/include-interceptor[0.2.4, 0.2.5] from composer repo (https://repo.packagist.org) but infection/include-interceptor is the root package and cannot be modified. See https://getcomposer.org/dep-on-root for details and assistance.
```

File: composer.json [hunk 3]
```diff
@@ -12,15 +12,18 @@
     "require": {},
     "require-dev": {
         "friendsofphp/php-cs-fixer": "^2.16",
-        "infection/infection": "^0.15.0",
+        "infection/infection": "^0.19.0",
```
Comment (peter279k) [inline composer.json:15]: Upgrading the `infection/infection` to support `8.x` versions.

File: composer.json [hunk 4]
```diff
@@ -12,15 +12,18 @@
...
         "php-coveralls/php-coveralls": "^2.2",
         "phpstan/phpstan": "^0.12.8",
         "phpunit/phpunit": "^8.5",
         "vimeo/psalm": "^3.8"
     },
     "config": {
-        "sort-packages": true
+        "sort-packages": true,
+        "allow-plugins": {
+            "infection/extension-installer": true
```
Comment (peter279k) [inline composer.json:25]: When upgrading the `infection/infection`, it presents the following message:

```
infection/extension-installer contains a Composer plugin which is currently not in your allow-plugins config. See https://getcomposer.org/allow-plugins
Do you trust "infection/extension-installer" to execute code and wish to enable it now? (writes "allow-plugins" to composer.json) [y,n,d,?] y
```

**Comment by maks-rafalko** [comment]
Thank you @peter279k

**Comment by maks-rafalko** [comment]
hm.. seems like GH actions fails. Would you be able to check what's wrong?

https://github.com/infection/include-interceptor/actions/runs/3873489237/jobs/6603564494

**Comment by peter279k** [comment]
The error is as follows:

```
php vendor/bin/phpunit --coverage-clover=build/logs/clover.xml --group=default
PHPUnit 8.5.31 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.1.13 with PCOV 1.0.[11](https://github.com/infection/include-interceptor/actions/runs/3873489237/jobs/6603564494#step:7:12)
Configuration: /home/runner/work/include-interceptor/include-interceptor/phpunit.xml.dist
Random seed:   1673265140
Error:         This version of PHPUnit does not support code coverage on PHP 8

...S......F........                                               19 / 19 (100%)

Time: 145 ms, Memory: 8.00 MB

There was 1 failure:

1) Infection\Tests\StreamWrapper\IncludeInterceptorTest::test_it_not_intercepts_when_not_included
Failed asserting that 'text' contains "1".

/home/runner/work/include-interceptor/include-interceptor/tests/IncludeInterceptorTest.php:[12](https://github.com/infection/include-interceptor/actions/runs/3873489237/jobs/6603564494#step:7:13)3
```

The problem is about the wrong PHPUnit version?

**Pull request merged**